converter/src/main/java/gov/cms/qpp/conversion/model/error/ErrorCode.java

```diff
@@ -0,0 +1,166 @@
+	CPC_QUALITY_MEASURE_ID_STRATA_MISMATCH("Amount of stratifications %d does not meet expectations %d for %s measure (%s). Expected strata: %s", true),
+	IA_MEASURE_INCORRECT_CHILDREN_COUNT("Measure performed must have exactly one child."),
+	IA_MEASURE_INVALID_TYPE("A single measure performed value is required and must be either a Y or an N."),
+	MEASURE_PERFORMED_MISSING_AGGREGATE_COUNT("Measure performed must have exactly one Aggregate Count."),
+	MEASURE_DATA_VALUE_NOT_INTEGER("Measure data must be a positive integer value"),
+	CPC_NPI_TIN_COMBINATION_MISSING_NPI_TIN_COMBINATION("Must have at least one NPI/TIN combination"),
+	CPC_PERFORMANCE_PERIOD_START_JAN12017("Must be 01/01/2017"),
+	CPC_PERFORMANCE_PERIOD_END_DEC312017("Must be 12/31/2017"),
+	QUALITY_MEASURE_ID_MISSING_SINGLE_MEASURE_POPULATION("The measure reference results must have a single measure population"),
+	QUALITY_MEASURE_ID_MISSING_SINGLE_MEASURE_TYPE("The measure reference results must have a single measure type"),
+	QUALITY_MEASURE_ID_INCORRECT_UUID("The eCQM (electronic measure id: %s) requires a %s with the correct UUID of %s", true),
+	QUALITY_MEASURE_ID_INCORRECT_PERFORMANCE_UUID("The eCQM (electronic measure id: %s) has a %s with an incorrect UUID of %s", true),
+	QUALITY_MEASURE_ID_MISSING_SINGLE_PERFORMANCE_RATE("A Performance Rate must contain a single Performance Rate UUID");
+
+	private final String message;
+	private final boolean hasFormat;
+
+	ErrorCode(String message) {
+		this(message, false);
+	}
+
+	ErrorCode(String message, boolean hasFormat) {
+		this.message = message;
+		this.hasFormat = hasFormat;
+	}
+
+	/**
+	 * Gets the message associated with this error code
+	 */
+	public final String getMessage() {
+		return message;
+	}
+
+	/**
+	 * Self returning
+	 */
+	@Override
+	@Deprecated // no use calling this
+	public ErrorCode getErrorCode() {
+		return this;
+	}
+
+	/**
+	 * Creates a formatted version of this error code, or throws an exception
+	 *
+	 * @param arguments arguments to format with
+	 * @return the formatted version of this error code, or throws an exception if formatting is not supported.
+	 */
+	public LocalizedError format(Object... arguments) {
+		if (hasFormat) {
+			String formatted = String.format(message, arguments);
+			return new FormattedErrorCode(formatted);
+		}
+
+		throw new IllegalStateException(this + " does not support formatting");
+	}
+
+	private final class FormattedErrorCode implements LocalizedError {
+		private final String message;
+
+		FormattedErrorCode(String message) {
+			this.message = message;
+		}
+
+		@Override
+		public ErrorCode getErrorCode() {
+			return ErrorCode.this;
+		}
+
+		@Override
+		public String getMessage() {
+			return message;
+		}
+
+		@Override
+		public String toString() {
+			return MoreObjects.toStringHelper(this)
+					.add("errorCode", ErrorCode.this)
+					.add("message", message)
+					.toString();
+		}
+
+		@Override
+		public boolean equals(Object o) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)